File: app/src/main/java/vn/loitp/app/activity/customviews/button/shinebutton/_lib/LShineView.java
package vn.loitp.app.activity.customviews.button.shinebutton._lib;

import android.content.Context;
import android.util.AttributeSet;
import android.util.TypedValue;
import android.view.View;
import android.widget.ImageView;
import android.widget.RelativeLayout;

import com.daimajia.androidanimations.library.Techniques;

import vn.loitp.app.utilities.LAnimationUtil;
import vn.loitp.app.utilities.LUIUtil;
import vn.loitp.livestar.R;

/**
 * Created by www.muathu@gmail.com on 5/13/2017.
 */

public class LShineView extends RelativeLayout {
    //guide: https://github.com/ChadCSong/ShineButton

    private final String TAG = getClass().getSimpleName();
    private ImageView iv;
    private ShineButton shineButton;

    public LShineView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public LShineView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init();
    }

    private void init() {
        inflate(getContext(), R.layout.view_l_shine, this);

        this.iv = (ImageView) findViewById(R.id.iv);
        this.shineButton = (ShineButton) findViewById(R.id.shine_button);
        iv.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                shineButton.performClick();
                LAnimationUtil.play(iv, Techniques.Pulse);
                if (callback != null) {
                    if (isDelay) {
                        LUIUtil.setDelay(750, new LUIUtil.DelayCallback() {
                            @Override
                            public void doAfter(int mls) {
                                callback.onClick(v);
                            }
                        });
                    } else {
                        callback.onClick(v);
                    }
                }
            }
        });
    }

    public void setSize(int sizeImageViewInDP, int sizeShineButtonInDP) {
        int dimensionInDpImageView = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, sizeImageViewInDP, getResources().getDisplayMetrics());
        iv.getLayoutParams().height = dimensionInDpImageView;
        iv.getLayoutParams().width = dimensionInDpImageView;
        iv.requestLayout();

        int dimensionInDpShineButton = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, sizeShineButtonInDP, getResources().getDisplayMetrics());
        shineButton.getLayoutParams().height = dimensionInDpShineButton;
        shineButton.getLayoutParams().width = dimensionInDpShineButton;
        shineButton.requestLayout();
    }

    public interface Callback {
        public void onClick(View view);
    }

    private boolean isDelay = true;

    public void setDelay(boolean isDelay) {
        this.isDelay = isDelay;
    }

    private Callback callback;

    public void setOnClick(Callback callback) {
        this.callback = callback;
    }

    public void setImage(int resID) {
        iv.setImageResource(resID);
    }

    public void setEnabledIv(boolean isEnable) {
        iv.setEnabled(isEnable);
    }

    public void setPerformClick() {
        iv.performClick();
    }
}

File: core/src/main/java/com/views/button/shinebutton/LShineView.kt
package com.views.button.shinebutton

import android.content.Context
import android.util.AttributeSet
import android.util.TypedValue
import android.view.View
import android.widget.ImageView
import android.widget.RelativeLayout
import com.core.utilities.LAnimationUtil
import com.core.utilities.LUIUtil
import com.daimajia.androidanimations.library.Techniques
import loitp.core.R

/**
 * Created by www.muathu@gmail.com on 5/13/2017.
 */

//guide: https://github.com/ChadCSong/ShineButton
class LShineView : RelativeLayout {
    private val TAG = javaClass.simpleName
    private var iv: ImageView? = null
    private var shineButton: ShineButton? = null

    private var isDelay = true

    private var callback: Callback? = null

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {
        init()
    }

    private fun init() {
        View.inflate(context, R.layout.view_l_shine, this)

        this.iv = findViewById(R.id.iv)
        this.shineButton = findViewById(R.id.shine_button)
        iv?.setOnClickListener { v ->
            shineButton?.performClick()
            LAnimationUtil.play(iv, Techniques.Pulse)
            if (callback != null) {
                if (isDelay) {
                    LUIUtil.setDelay(750) { mls -> callback?.onClick(v) }
                } else {
                    callback?.onClick(v)
                }
            }
        }
    }

    fun setSize(sizeImageViewInDP: Int, sizeShineButtonInDP: Int) {
        val dimensionInDpImageView = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, sizeImageViewInDP.toFloat(), resources.displayMetrics).toInt()
        iv?.let {
            it.layoutParams.height = dimensionInDpImageView
            it.layoutParams.width = dimensionInDpImageView
            it.requestLayout()
        }

        val dimensionInDpShineButton = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, sizeShineButtonInDP.toFloat(), resources.displayMetrics).toInt()
        shineButton?.let {
            it.layoutParams?.height = dimensionInDpShineButton
            it.layoutParams?.width = dimensionInDpShineButton
            it.requestLayout()
        }
    }

    interface Callback {
        fun onClick(view: View)
    }

    fun setDelay(isDelay: Boolean) {
        this.isDelay = isDelay
    }

    fun setOnClick(callback: Callback) {
        this.callback = callback
    }

    fun setImage(resID: Int) {
        iv?.setImageResource(resID)
    }

    fun setEnabledIv(isEnable: Boolean) {
        iv?.isEnabled = isEnable
    }

    fun setPerformClick() {
        iv?.performClick()
    }
}